core/internal/leet/model_update_test.go

```diff
@@ -0,0 +1,524 @@
+
+	model.TestHandleChartGridClick(0, 0)
+	require.Equal(t, leet.FocusNone, model.TestFocusState().Type)
+
+	model.TestHandleChartGridClick(0, 1)
+	require.Equal(t, leet.FocusMainChart, model.TestFocusState().Type)
+
+	model.TestClearMainChartFocus()
+	require.Equal(t, leet.FocusNone, model.TestFocusState().Type)
+}
+
+func TestHandleOverviewFilter_TypingSpaceBackspaceEnterEsc(t *testing.T) {
+	logger := observability.NewNoOpLogger()
+	cfg := leet.NewConfigManager(filepath.Join(t.TempDir(), "config.json"), logger)
+	var m tea.Model = leet.NewModel("dummy", cfg, logger)
+	m, _ = m.Update(tea.WindowSizeMsg{Width: 180, Height: 60})
+
+	// Enter overview filter mode
+	m, _ = m.Update(tea.KeyMsg{Type: tea.KeyRunes, Runes: []rune{'o'}})
+
+	// Type "acc", add space, backspace, then Enter
+	m, _ = m.Update(tea.KeyMsg{Type: tea.KeyRunes, Runes: []rune("ac")})
+	m, _ = m.Update(tea.KeyMsg{Type: tea.KeyRunes, Runes: []rune("c")})
+	m, _ = m.Update(tea.KeyMsg{Type: tea.KeySpace})
+	m, _ = m.Update(tea.KeyMsg{Type: tea.KeyBackspace})
+	m, _ = m.Update(tea.KeyMsg{Type: tea.KeyEnter})
+
+	model := m.(*leet.Model)
+	require.True(t, model.TestSidebarIsFiltering())
+	require.Equal(t, "acc", model.TestSidebarFilterQuery())
+
+	// Enter filter mode again, type something, then Esc
+	m, _ = m.Update(tea.KeyMsg{Type: tea.KeyRunes, Runes: []rune{'o'}})
+	m, _ = m.Update(tea.KeyMsg{Type: tea.KeyRunes, Runes: []rune("tmp")})
+	_, _ = m.Update(tea.KeyMsg{Type: tea.KeyEsc})
+
+	// Should keep the previously applied "acc" state
+	require.True(t, model.TestSidebarIsFiltering())
+	require.Equal(t, "acc", model.TestSidebarFilterQuery())
+}
+
+func TestHandleKeyMsg_VariousPaths(t *testing.T) {
+	logger := observability.NewNoOpLogger()
+	cfg := leet.NewConfigManager(filepath.Join(t.TempDir(), "config.json"), logger)
+	var m tea.Model = leet.NewModel("dummy", cfg, logger)
+	m, _ = m.Update(tea.WindowSizeMsg{Width: 180, Height: 50})
+
+	// Toggle left sidebar
+	m, _ = m.Update(tea.KeyMsg{Runes: []rune{'['}, Type: tea.KeyRunes})
+	model := m.(*leet.Model)
+
+	// Force complete the animation
+	ls := model.TestGetLeftSidebar()
+	ls.TestForceExpand()
+
+	require.True(t, model.TestLeftSidebarVisible())
+
+	// Page navigation (shouldn't panic)
+	m, _ = m.Update(tea.KeyMsg{Type: tea.KeyPgUp})
+	m, _ = m.Update(tea.KeyMsg{Type: tea.KeyShiftUp})
+	m, _ = m.Update(tea.KeyMsg{Type: tea.KeyPgDown})
+	m, _ = m.Update(tea.KeyMsg{Type: tea.KeyShiftDown})
+
+	// Help toggle
+	m, _ = m.Update(tea.KeyMsg{Runes: []rune{'h'}, Type: tea.KeyRunes})
+	m, _ = m.Update(tea.KeyMsg{Runes: []rune{'?'}, Type: tea.KeyRunes})
+
+	// Overview filter
+	m, _ = m.Update(tea.KeyMsg{Runes: []rune{'['}, Type: tea.KeyRunes})
+	m, _ = m.Update(tea.KeyMsg{Type: tea.KeySpace})
+	m, _ = m.Update(tea.KeyMsg{Type: tea.KeyBackspace})
+	m, _ = m.Update(tea.KeyMsg{Type: tea.KeyEnter})
+
+	// Clear overview filter
+	_, _ = m.Update(tea.KeyMsg{Type: tea.KeyCtrlK})
+}
+
+func TestHeartbeat_LiveRun(t *testing.T) {
```

Comment:
Since we run tests automatically on `git push`, the multiple sleeping seconds here will be disruptive.

If you have to keep it, maybe you could add the `-short` flag to `core/scripts/pre-commit-hooks/run-go-unit-tests.sh` and then skip the test if `testing.Short()` is true [like this](https://pkg.go.dev/testing#hdr-Skipping).